### PR TITLE
Unattended gpg and alternate keyrings

### PR DESCRIPTION
**Pull request by stickycode**
I've been testing kubesec as part of monorepo projection tooling https://github.com/branchout/branchout and needed to be able to use a passphrase for keys supplied on the command line (and automated), turns out you can avoid pinentry with a loopback which makes testing a lot easier.

The alternate keyring support almost worked but failed to lookup primary keys for decryption

### CODE REVIEW

File: gpg/gpg.go [hunk 1]
```diff
@@ -297,6 +307,9 @@ func executeInShell(command ...string) error {
 }
 
 func executeInShellAndGrabOutput(command ...string) ([]byte, error) {
+	if keyring != "" {
+		command = append(command, "--no-default-keyring", "--keyring", keyring)
```
Comment (shyiko) [inline gpg/gpg.go]: any chance you can move this out of executeInShellAndGrabOutput?

Comment (stickycode) [reply gpg/gpg.go]: yes I guess so

Comment (stickycode) [reply gpg/gpg.go]: Ok thats cleaner and general

**Comment by stickycode** [comment]
@shyiko Is that looking better?

**Comment by shyiko** [comment]
It's perfect. Thank you!

**Pull request merged**